conformance/utils/kubernetes/helpers.go

```diff
@@ -626,8 +627,19 @@ func listenersMatch(t *testing.T, expected, actual []v1beta1.ListenerStatus) boo
 			t.Logf("Name doesn't match")
 			return false
 		}
-		if !reflect.DeepEqual(aListener.SupportedKinds, eListener.SupportedKinds) {
-			t.Logf("Expected SupportedKinds to be %v, got %v", eListener.SupportedKinds, aListener.SupportedKinds)
+
+		if len(eListener.SupportedKinds) == 0 && len(aListener.SupportedKinds) != 0 {
+			t.Logf("Expected list of SupportedKinds was empty, but the actual list for comparison was not:  %v",
+				aListener.SupportedKinds)
+			return false
+		}
+		// Ensure that the expected Listener.SupportedKinds items are present in actual Listener.SupportedKinds
+		// Find the items instead of performing an exact match of the slice because the implementation
+		// might support more Kinds than defined in the test
+		eSupportedKindsSet := sets.New(eListener.SupportedKinds...)
+		aSupportedKindsSet := sets.New(aListener.SupportedKinds...)
+		if !aSupportedKindsSet.IsSuperset(eSupportedKindsSet) {
+			t.Logf("Expected %v kinds to be present in SupportedKinds", eSupportedKindsSet.Difference(aSupportedKindsSet))
```

Comment:
I'm hitting an issue with this updated code, looks like it's doing pointer comparison on the Group string pointer whereas the old code was using reflect.DeepEqual so doing value comparison?

Reply:
Created https://github.com/kubernetes-sigs/gateway-api/issues/1703 for this thank you!

Reply:
yup thanks for finding this @skriss